### ISSUE TITLE
Add an exception for no-unused-expression for quick null-checking

### ISSUE THREAD

**Issue by orkisz** [palantir/tslint#1570]
### Bug Report

- __TSLint version__: `3.15.1`
- __TypeScript version__: `2.0.2`
- __Running TSLint via__: CLI / WebStorm

#### TypeScript code being linted

```ts
  public ngOnDestroy(): void {
    this._indicatorsDataSubscription && this._indicatorsDataSubscription.unsubscribe();
  }
```

with `tslint.json` configuration:

```json
    "no-unused-expression": true,
```

#### Actual behavior

TSLint reports left side of && operator as an unused expression.

#### Expected behavior

It would be nice to implement it as an exception. Otherwise user is forced to create `if` statements for those kind of disposals (especially annoying for disposing observables).

**Comment by IllusionMH** [comment]
Duplicate of #1413 

I've started to work on implementation, but postponed it because of vacation.
At this moment I have work in progress implementation that works on such simple example, but have false positives and false negatives in more complex cases.

**Comment by orkisz** [comment]
Right, I had missed it. So my issue can be closed then, I will monitor #1413
